### PR TITLE
Add Social Sharing component

### PR DESCRIPTION
**Pull request by matijs**
image

- [x] Add icon
- [x] Add a sensible label
- [x] social sharing site specific sharing behaviour
  - [x] Twitter
  - [x] Facebook
  - [x] ~Instagram~
  - [x] Linkedin

### CODE REVIEW

File: src/js/components/molecules/social-sharing/social-sharing.ts [hunk 1]
```diff
@@ -0,0 +1,44 @@
+/** @format */
+
+import { Vue, Component, Prop } from 'vue-property-decorator';
+// import { SocialSharingProps } from './social-sharing.models';
```
Comment (zwaardje) [inline src/js/components/molecules/social-sharing/social-sharing.ts]: Why are you not using the props?

File: src/js/components/molecules/social-sharing/social-sharing.ts [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+import { generateSocialSharingUrl } from '../../../utils/generate-social-sharing-url';
+import BrowserCapabilities from '../../../utils/browser-capabilities';
+
+@WithRender
+@Component({
+    name: 'SocialSharing',
+})
+
+export class SocialSharing extends Vue {
+    // @Prop({ type: Array, required: true }) items!: SocialSharingProps['items'];
```
Comment (zwaardje) [inline src/js/components/molecules/social-sharing/social-sharing.ts]: Use the props

File: src/js/components/molecules/social-sharing/social-sharing.ts [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+    get items() {
```
Comment (zwaardje) [inline src/js/components/molecules/social-sharing/social-sharing.ts]: Why? They are in items, right?

File: src/js/components/molecules/social-sharing/social-sharing.ts [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+            },
+            {
+                type: 'twitter',
+                iconName: 'twitter',
+                label: 'Share on Twitter',
+            },
+        ];
+    }
+
+    url({ type }) {
```
Comment (zwaardje) [inline src/js/components/molecules/social-sharing/social-sharing.ts]: Not using the $route?

File: src/js/components/molecules/social-sharing/social-sharing.ts (hunk 2 above)
Comment (matijs) [reply src/js/components/molecules/social-sharing/social-sharing.ts]: There are no props… it's hardcoded… for now.

File: src/js/components/molecules/social-sharing/social-sharing.ts (hunk 3 above)
Comment (matijs) [reply src/js/components/molecules/social-sharing/social-sharing.ts]: Nope… hardcoded :(

File: src/js/components/molecules/social-sharing/social-sharing.ts (hunk 4 above)
Comment (matijs) [reply src/js/components/molecules/social-sharing/social-sharing.ts]: Nope… did not seem to work… besides the `$route` only provided `fullPath` and would have to be prepended with the protocol and hostname.

File: src/js/components/molecules/social-sharing/social-sharing.ts (hunk 1 above)
Comment (matijs) [reply src/js/components/molecules/social-sharing/social-sharing.ts]: There are no props… it's hardcoded… for now.

Comment (zwaardje) [reply src/js/components/molecules/social-sharing/social-sharing.ts]: Ok cool,l it not managed in the CMS, then please remove the line given it's not needed.

File: src/js/components/molecules/social-sharing/social-sharing.ts (hunk 2 above)
Comment (zwaardje) [reply src/js/components/molecules/social-sharing/social-sharing.ts]: Ok cool,l it not managed in the CMS, then please remove the line given it's not needed.

File: src/js/components/molecules/social-sharing/social-sharing.ts (hunk 4 above)
Comment (zwaardje) [reply src/js/components/molecules/social-sharing/social-sharing.ts]: Ok then its fine

**Pull request merged**